pkg/controller/operators/catalog/operator.go

```diff
@@ -1331,10 +1331,43 @@ func (o *Operator) syncInstallPlans(obj interface{}) (syncError error) {
 		return
 	}
 
+	if plan.Status.Phase == v1alpha1.InstallPlanPhaseFailed {
+		return
+	}
+
 	querier := o.serviceAccountQuerier.NamespaceQuerier(plan.GetNamespace())
 	ref, err := querier()
 	if err != nil {
-		syncError = fmt.Errorf("attenuated service account query failed - %v", err)
+
+		// Retry sync if non-fatal error
+		if !scoped.IsOperatorGroupError(err) {
+			syncError = fmt.Errorf("attenuated service account query failed - %v", err)
+			return
+		}
+
+		// Mark the InstallPlan as failed for a fatal Operator Group related error
+		logger.Infof("attenuated service account query failed - %v", err)
+		ipFailError := fmt.Errorf("invalid operator group - %v", err)
+		now := o.now()
+		out := plan.DeepCopy()
+		out.Status.SetCondition(v1alpha1.ConditionFailed(v1alpha1.InstallPlanInstalled,
```

Comment:
This could result in a Complete -> Failed transition, which seems confusing. I think you'll either need to expand the values of Phase that are considered terminal above or somehow push more of this into transitionInstallPlanState to avoid that.

Reply:
I think I would rather consider `Complete` as terminal as well, similar to `Failed` and just end the sync early.
For all intents and purposes `Complete` is treated as terminal anyway right now by the InstallPlan sync, or more specifically [`transitionInstallPlanState()`](https://github.com/operator-framework/operator-lifecycle-manager/blob/bc37ac4e271272a8f8fbac4d1b08157b4883ec9f/pkg/controller/operators/catalog/operator.go#L1495-L1497).
That's more clear and separate to me than trying to handle all of this in `transitionInstallPlanState()` and passing it the results of the NamespaceQuerier.

Although please correct me if there's ever a case where we need to reconcile something more after the InstallPlan hits `Complete`.

Reply:
As far as I know the InstallPlan is effectively a historical record once it has reached either Failed or Complete. Returning early in both cases makes sense to me (feels related to https://medium.com/@matryer/line-of-sight-in-code-186dd7cdea88).

Reply:
TIL the official term for "return early to reduce nesting".